### PR TITLE
For #19886 - Add favicon to website info view

### PR DESCRIPTION
**Pull request by gabrielluong**
Co-authored-by: Arturo Mejia <arturomejiamarmol@gmail.com>

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/layout/quicksettings_website_info.xml [hunk 1]
```diff
@@ -15,11 +15,21 @@
...
+            android:layout_height="24dp"
+            android:importantForAccessibility="no"
+            android:scaleType="fitCenter"
+            android:layout_marginStart="16dp"
+            android:layout_gravity="center_vertical"
+            tools:drawableStartCompat="@drawable/ic_internet" />
+
         <TextView
             android:id="@+id/url"
             style="@style/QuickSettingsText"
```
Comment (Amejia481) [inline app/src/main/res/layout/quicksettings_website_info.xml:35]: We also will need to add `app:drawableEndCompat="@drawable/ic_arrowhead_right"` and `android:paddingEnd="0dp"` to add the arrow icon and align the icon with the **"Blocked items"**'s arrow section.

Comment (Amejia481) [reply app/src/main/res/layout/quicksettings_website_info.xml:35]: I will add that

Comment (Amejia481) [reply app/src/main/res/layout/quicksettings_website_info.xml:35]: See https://github.com/mozilla-mobile/fenix/pull/20144/commits/6ea5281dab40ec4762cbc3fbbe69af839d936efd#diff-e4bce07500a19f5ea4f0a09151a4ff7a0a5d823e6dfe4ae5cb3ecab6d4662524R66

**Comment by Amejia481** [comment]
There are also some tests that are failing on `run-testDebugUnitTest` I will take a look

File: app/src/main/res/layout/quicksettings_website_info.xml (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/res/layout/quicksettings_website_info.xml:35]: Ya that was intentional. I was just adding the favicon here.

**Review by Amejia481 — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/settings/quicksettings/WebsiteInfoViewTest.kt [hunk 2]
```diff
@@ -18,21 +23,28 @@ import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
...
-            websiteUrl = "https://mozilla.org",
+            websiteUrl = websiteUrl,
             websiteTitle = "Mozilla",
             websiteSecurityUiValues = WebsiteSecurityUiValues.SECURE,
             certificateName = ""
         ))
 
+        verify { icons.loadIntoView(view.favicon_image, IconRequest(websiteUrl)) }
+
         assertEquals("https://mozilla.org", view.url.text)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/settings/quicksettings/WebsiteInfoViewTest.kt:48]:
```suggestion
        assertEquals("mozilla.org", view.url.text)
```

Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/settings/quicksettings/WebsiteInfoViewTest.kt]: updated

**Comment by gabrielluong** [comment]
https://github.com/mozilla-mobile/fenix/pull/20255/checks?check_run_id=2992766840 testDebugUnitTest failures are for metrics ping and unrelated to our changes. Going ahead with landing.

**Pull request merged**